src/main.rs

```diff
@@ -211,16 +211,45 @@ struct Args {
     )]
     oci_password: Option<String>,
 
-    /// Specifies which exporter to use for traces. Only "otlp" is supported at this time
-    #[clap(long = "otel-traces-exporter", env = "OTEL_TRACES_EXPORTER")]
-    otel_traces_exporter: Option<String>,
+    /// Determines whether observability should be enabled.
+    #[clap(long = "enable-observability", env = "WASMCLOUD_OBSERVABILITY_ENABLED")]
+    enable_observability: bool,
 
-    /// Specifies the endpoint to use for the OTLP exporter
+    /// Determines whether tracing should be enabled, overriding any value set with `--enable-observability`.
     #[clap(
-        long = "otel-exporter-otlp-endpoint",
-        env = "OTEL_EXPORTER_OTLP_ENDPOINT"
+        long = "enable-tracing",
+        env = "WASMCLOUD_TRACING_ENABLED",
+        hide = true
+    )]
+    enable_tracing: Option<bool>,
+
+    /// Determines whether metrics should be enabled, overriding any value set with `--enable-observability`.
+    #[clap(
+        long = "enable-metrics",
+        env = "WASMCLOUD_METRICS_ENABLED",
+        hide = true
     )]
-    otel_exporter_otlp_endpoint: Option<String>,
+    enable_metrics: Option<bool>,
+
+    /// Determines whether logs should be enabled, overriding any value set with `--enable-observability`.
+    #[clap(long = "enable-logs", env = "WASMCLOUD_LOGS_ENABLED", hide = true)]
+    enable_logs: Option<bool>,
+
+    /// Overrides the OpenTelemetry endpoint used for emitting traces, metrics and logs. This can also be set with `OTEL_EXPORTER_OTLP_ENDPOINT`.
```

Comment:
👨🏻‍🍳 💋

Reply:
As much as I'd love to keep this comment around, it looks like I need to resolve it to appease the GitHub machinery 😆